powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/ActivationService.java

```diff
@@ -421,6 +352,149 @@ public ActivationLayer1Response createActivation(ActivationLayer1Request request
+                httpCustomizationService.getHttpHeaders()
+        );
+
+        final String activationId = response.getActivationId();
+        final String applicationId = response.getApplicationId();
+
+        final UserInfoContext userInfoContext = UserInfoContext.builder()
+                .stage(UserInfoStage.ACTIVATION_PROCESS_CUSTOM)
+                .userId(userId)
+                .activationId(activationId)
+                .applicationId(applicationId)
+                .build();
+        final Map<String, Object> userInfo = processUserInfo(userInfoContext);
+
+        // Process custom attributes using a custom logic
+        final Map<String, Object> processedCustomAttributes = activationProvider.processCustomActivationAttributes(customAttributes, activationId, userId, applicationId, ActivationType.CUSTOM, context);
+
+        // Save activation flags in case the provider specified any flags
+        final List<String> activationFlags = activationProvider.getActivationFlags(identity, processedCustomAttributes, activationId, userId, applicationId, ActivationType.CUSTOM, context);
+        if (activationFlags != null && !activationFlags.isEmpty()) {
+            final AddActivationFlagsRequest flagsRequest = new AddActivationFlagsRequest();
+            flagsRequest.setActivationId(activationId);
+            flagsRequest.getActivationFlags().addAll(activationFlags);
+            powerAuthClient.addActivationFlags(
+                    flagsRequest,
+                    httpCustomizationService.getQueryParams(),
+                    httpCustomizationService.getHttpHeaders()
+            );
+        }
+
+        // Check if activation should be committed instantly and if yes, perform commit
+        if (activationProvider.shouldAutoCommitActivation(identity, customAttributes, activationId, userId, applicationId, ActivationType.CUSTOM, context)) {
+            final CommitActivationResponse commitResponse = commitActivation(activationId);
+            if (commitResponse.isActivated()) {
+                activationProvider.activationWasCommitted(identity, customAttributes, activationId, userId, applicationId, ActivationType.CUSTOM, context);
+            }
+        }
+
+        // Prepare encrypted activation data
+        return prepareEncryptedResponse(response.getEncryptedData(), response.getMac(),
+                response.getNonce(), response.getTimestamp(), processedCustomAttributes, userInfo);
+    }
+
+    private ActivationLayer1Response processOidcActivation(final EncryptionContext eciesContext, final ActivationLayer1Request request) throws PowerAuthClientException, PowerAuthActivationException {
+        logger.debug("Processing direct OIDC activation.");
+
+        final Map<String, String> identity = request.getIdentityAttributes();
+        final OAuth2ActivationContext oAuthActivationContext = OAuth2ActivationContext.builder()
+                .clientId(identity.get("clientId"))
```

Comment:
@Hopsaheysa @kober32 In TLDR, we have identification via `clientId` but I suggest using `providerId` to make it consistent with the configuration endpoint introduced in https://github.com/wultra/enrollment-server/pull/1107 Any opinion?

Reply:
Sounds good to me.